### ISSUE TITLE
Compatibility issue with Alternate Screen Buffer and [console]::CursorVisible

### ISSUE THREAD

**Issue by jjcarrier** [PowerShell/PowerShell#21212]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Below is a script that illustrates the issue. When using the alternate screen buffer, the statement `[console]::CursorVisible = $false` does not function correctly. This of course can be remedied by using the ANSI sequence instead. It is not clear if this is by design or not.

```pwsh
function Enter-AltScreenBuffer
{
    $Host.UI.Write([char]27 + "[?1049h")
}

function Exit-AltScreenBuffer
{
    $Host.UI.Write([char]27 + "[?1049l")
}

function Show-TerminalCursor
{
    $Host.UI.Write([char]27 +"[?25h")
}

function Hide-TerminalCursor
{
    $Host.UI.Write([char]27 +"[?25l")
}

Clear-Host

# This is the working case, using the associated ANSI sequence.
Write-Output "Test 1"

Enter-AltScreenBuffer
Hide-TerminalCursor
Clear-Host
Write-Output "Alternate Buffer: Cursor Hidden (Working)"
while ($Host.ui.RawUI.ReadKey().VirtualKeyCode -ne [ConsoleKey]::Enter) {
    $Host.UI.Write("`r `r")
}
Show-TerminalCursor
Exit-AltScreenBuffer

# This is the non-working case, using the [console]::CursorVisible.
Write-Output "Test 2"

Enter-AltScreenBuffer
[console]::CursorVisible = $false
Clear-Host
Write-Output "Alternate Buffer: Cursor Hidden (Broken)"
while ($Host.ui.RawUI.ReadKey().VirtualKeyCode -ne [ConsoleKey]::Enter) {
    $Host.UI.Write("`r `r")
}
[console]::CursorVisible = $true
Exit-AltScreenBuffer
```

### Expected behavior

```console
When setting `[console]::CursorVisible = $false` the cursor is expected to be hidden.
```

### Actual behavior

```console
When executing `[console]::CursorVisible = $false` within the alternate screen buffer, the cursor remains visible.
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.1
PSEdition                      Core
GitCommitId                    7.4.1
OS                             Microsoft Windows 10.0.22631
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by 237dmitry** [comment]
I could not reproduce your example. The cursor is hidden as expected.

**Comment by jjcarrier** [comment]
Interesting, I can reliably reproduce this on three different computers (two on Win11, one on Win10).
I am able to reproduce on both Windows Terminal and Windows Console Host.
For what it is worth I have, Windows Terminal: v1.19.10302.0.
I also realized I can reproduce this on Windows PowerShell.

**Comment by mklement0** [comment]
I also cannot reproduce this, although I only tried on one machine (W11 22H2).

While my WT version is older (and asking for it to updated finds no newer version, even though it clearly exists), everything works as expected, including in `conhost.exe` windows and in Windows PowerShell (as well as on macOS).

Also, whether or not VT support is enabled system-wide (value `VirtualTerminalLevel` in registry key `HKEY_CURRENT_USER\Console`) makes no difference for me (which is to be expected, given that PowerShell unconditionally enables this support for itself, except when calling external programs, in which case it defers to the registry setting).

**Comment by daxian-dbw** [comment]
I cannot reproduce either. I'm on Win11 with WT version 1.20.10303.0.
Also, this is not a PowerShell issue, as `[console]::CursorVisible` is a .NET API. If you can reproduce reliably, then maybe try coming up with a .NET application to reproduce the issue, and then open an issue in the dotnet/runtime repo.

**Comment by jjcarrier** [comment]
I ran some more tests but sadly have not learned much from the info gathered, just reporting it here in case it becomes useful later.

I _also_ cannot reproduce the issue on macOS or Linux (including Ubuntu/Debian/Raspbian and WSL Ubuntu). Both WSL Ubuntu and native Ubuntu were tested on two of the Win11 PCs observed to have this issue.

I also tried other terminals:
- Windows Terminal Preview 1.20.10303.0: Issue present
- VSCode terminal: Issue present
- Cmder: Issue not present
  -  I did notice that this emulator uses `conhost.exe` version `10.0.22621.2506` while WT is using `OpenConsole.exe` version `1.19.2401.30002`. But this does not mean much to me since I have seen the issue on both underlying console hosts.

> If you can reproduce reliably, then maybe try coming up with a .NET application to reproduce the issue, and then open an issue in the dotnet/runtime repo.

Will do. I will close this issue.
